src/common/xo-week-heatmap/index.js

```diff
@@ -44,66 +67,66 @@ export default class XoWeekHeatmap extends Component {
   }
 
   componentWillReceiveProps (nextProps) {
-    this._updateState(nextProps)
+    this._updateDays(nextProps)
   }
 
   componentWillMount () {
-    this._updateState()
+    this._updateDays()
   }
 
-  _computeColorGen = data => scaleSequential(interpolateViridis)
-    .domain(extent(data, value => value.value).reverse())
-
-  _updateState (props = this.props) {
+  _updateDays (props = this.props) {
     const {
       data,
       intl: {
         formatTime
       }
     } = props
-    const colorGen = this._computeColorGen(props.data)
-    const matrix = {}
+    const days = {}
 
+    // 1. Compute average per day.
     forEach(data, elem => {
       const date = new Date(elem.date)
-      const day = date.getDate()
-      const hour = date.getHours()
+      const monthDay = formatTime(date, { day: '2-digit' })
+      const dayId = `${formatTime(date, { month: '2-digit' })}${monthDay}`
```

Comment:
Do not use `formatTime` here, compute the key differently (`YYYYMMdd`) use moment if necessary.